### PR TITLE
Upgrade to ton v13 and sandbox

### PR DESCRIPTION
**Pull request by hrmon**

### CODE REVIEW

File: build/_build.ts [hunk 1]
```diff
@@ -31,9 +31,12 @@ async function main() {
       .execSync("func -V")
```
Comment (shaharyakir) [inline build/_build.ts:31]: I think we wanted to use blueprint, which should make all scripts under `build/` obsolete

File: wrappers/verifier-registry.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
+import {
```
Comment (shaharyakir) [inline wrappers/verifier-registry.ts:1]: probably we should adapt the Contract interface for this wrapper (similar to what was applied to Sources Registry)?

File: contracts/sources-registry.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
-import BN, { min } from "bn.js";
 import { Cell, beginCell, Address } from "ton";
```
Comment (shaharyakir) [inline contracts/sources-registry.ts:1]: I think this also could become obsolete if we adapt blueprint

File: test/e2e/makeGetCall.ts [hunk 4]
```diff
@@ -1,40 +1,49 @@
...
       default:
-        throw new Error(`unknown type: ${type}, val: ${JSON.stringify(val)}`);
+        throw new Error(`unknown type: ${item.type}`);
     }
-  });
+  }
+  return parsedItems;
 }
 
 export async function makeGetCall<T>(
```
Comment (shaharyakir) [inline test/e2e/makeGetCall.ts:49]: is `makeGetCall` still used anywhere after adapting sandbox?

**Review by mrbonezy — COMMENTED** [review]
See comments

File: build/_build.ts (hunk 1 above)
Comment (hrmon) [reply build/_build.ts:31]: Right. old build scripts use system binaries for compiling contracts. So we should switch to func-js? which version?

File: wrappers/verifier-registry.ts (hunk 2 above)
Comment (hrmon) [reply wrappers/verifier-registry.ts:1]: It does implement the Contract interface. Do you mean the queries function should be integrated into the contract class?

File: test/e2e/makeGetCall.ts (hunk 4 above)
Comment (hrmon) [reply test/e2e/makeGetCall.ts:49]: There is this `sources-registry-only-set-code.fc` which is used in the tests. It has a `am_i_replaced` get method which is called in those test. Should we add another  wrapper for this dummy contract?

File: build/_build.ts (hunk 1 above)
Comment (shaharyakir) [reply build/_build.ts:31]: Yes, we can use the version blueprint comes with

File: wrappers/verifier-registry.ts (hunk 2 above)
Comment (shaharyakir) [reply wrappers/verifier-registry.ts:1]: More in the sense that anything using VerifierRegistry shouldn't be aware of Queries.
There should be instead functions for each operation in VerifierRegistry's API.

File: test/e2e/makeGetCall.ts (hunk 4 above)
Comment (shaharyakir) [reply test/e2e/makeGetCall.ts:49]: Yes, just so we can lost `makeGetCall` altogether

**Comment by shaharyakir** [comment]
Looks good, but I think we're missing deploy scripts for blueprint ([example](https://github.com/ton-org/blueprint/blob/main/example/scripts/deployCounter.ts))

**Comment by shaharyakir** [comment]
@hrmon CI fails - can you make sure that `npm run build` triggers `npx blueprint build` for all contracts

**Review by MrG9090 — APPROVED** [review]

**Review by MrG9090 — APPROVED** [review]

**Review by MrG9090 — APPROVED** [review]

**Review by MrG9090 — APPROVED** [review]

**Review by albertoblue87-netizen — COMMENTED** [review]
Done

**Review by albertoblue87-netizen — COMMENTED** [review]
Done

**Review by Vicinit — APPROVED** [review]

**Pull request merged**